### DISCUSSION TITLE
Allowing Free Styling

### DISCUSSION THREAD

**Discussion by jt6677** [realworld-apps/realworld#769] [Ideas]
I see tremendous value in enforcing strict API and styling spec. I did it for the frontend portion but tried it with TailwindCSS. I had to mock all the APIs with json-server locally. Strict guideline is a must in the beginning otherwise everything is in chaos. But as this project matures, I see little innovation. For the sake of learning, I hope you can consider open a different section something like "not comply with the spec and consume at your own risk".  I did this https://homebh.ga  https://github.com/jt6677/conduit-fullstack     work-in-progress,  it is just for learning ts.

**Comment by geromegrignon** [comment]
Hey @jt6677, the next step for this project is to work on the v2 for more modern specs (as the current specs were created  6 years ago).
Part of these changes is to allow CSS frameworks as such choice is fully part of modern real-world considerations.
Besides such change, we'll encourage the community to challenge the current design of the application.

But we won't allow "not comply with the spec and consume at your own risk" category.
I'm happy this project is an opportunity for you to learn more about Typescript but please consider everyone's journey is unique and you can't expect everyone to understand your application just because it exists: that's the point of the specs.
Even by sticking with specs they are so many interesting ways to approach a solution. But thanks to these specs, there is a common point to understand them without prerequisites.
